apps/autoscheduler/CMakeLists.txt

```diff
@@ -0,0 +1,169 @@
+# weights
+file(GLOB WF baseline.weights)
+get_filename_component(BASENAME_WF ${WF} NAME_WE)
+set(WF_CPP ${BASENAME_WF}.cpp)
+
+add_custom_command(OUTPUT ${WF_CPP}
+                   DEPENDS ${WF} binary2cpp
+                   COMMAND binary2cpp ${BASENAME_WF}_weights < ${WF} > ${WF_CPP}
+                   COMMENT "${WF} -> ${WF_CPP}"
+                   VERBATIM)
+
+# =======================================================
+
+# retrain_cost_model
+add_executable(retrain_cost_model_process
+               ASLog.cpp
+               DefaultCostModel.cpp
+               Weights.cpp
+               retrain_cost_model.cpp
+               ${WF_CPP})
+set_target_properties(retrain_cost_model_process
+                      PROPERTIES OUTPUT_NAME retrain_cost_model)
+
+set(COST_MODEL_GENERATOR_FILES cost_model_generator.cpp)
+halide_generator(cost_model.generator SRCS ${COST_MODEL_GENERATOR_FILES})
+halide_generator(train_cost_model.generator SRCS ${COST_MODEL_GENERATOR_FILES})
```

Comment:
You shouldn't need to make a separate halide_generator for these; you should be able to just define cost_model.generator and invoke it with a separate GENERATOR argument (this is what we do in autoscheduler.inc)

Reply:
It seems to be impossible to do this without changing the `halide.cmake` file. Currently, `GENERATOR_NAME` is not a parameter but is a property. We can look at the example of the `linear_algebra/blas_generators` where are also creating a lot of generators.

Reply:
Yep, you are correct -- my bad here. I'll ponder whether making it configurable makes sense.